### PR TITLE
LPS-84618 UserIndexer migrate to new indexer architecture

### PR DESCRIPTION
**Pull request by luanmaoski**
This PR depends on [Contact](https://github.com/BryanEngler/liferay-portal/pull/255) migration.

### CODE REVIEW

**Comment by lmarqs** [comment]
reviewing

File: modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/BaseUserIndexerTestCase.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	}
+
+	protected UserFixture createUserFixture() {
+		return new UserFixture(userLocalService, _groups, _users);
+	}
+
+	protected UserIndexerFixture createUserIndexerFixture() {
+		Indexer<User> indexer = indexerRegistry.getIndexer(User.class);
+
+		return new UserIndexerFixture(indexer);
```
Comment (lmarqs) [inline modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/BaseUserIndexerTestCase.java]: @luanmaoski, `indexer` is being passed as null. Let's check together

File: modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerIndexedFieldsTest.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	private Map<String, String> _expectedFieldValues(User user)
+		throws Exception {
+
+		Map<String, String> map = new HashMap<>();
+
+		map.put(Field.ENTRY_CLASS_PK, String.valueOf(user.getUserId()));
+		map.put(Field.ENTRY_CLASS_NAME, User.class.getName());
+		map.put(Field.COMPANY_ID, String.valueOf(user.getCompanyId()));
+
+		map.put("firstName", StringUtil.toLowerCase(user.getFirstName()));
```
Comment (lmarqs) [inline modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerIndexedFieldsTest.java]: @luanmaoski, If I'm not wrong, only "*_sortable" fields are lower case.

File: modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerIndexedFieldsTest.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+			SynchronousDestinationTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		super.setUp();
+	}
+
+	@Test
+	public void testIndexedFields() throws Exception {
+		String firstName = "richard";
```
Comment (lmarqs) [inline modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerIndexedFieldsTest.java]: @luanmaoski, I think we should use some capital letters here, like `"Richard Bower"`.
Also we could use `RandomStringTestUtil.randomString()`.

File: modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerReindexTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+	@Test
+	public void testIndexedFields() throws Exception {
+		String firstName = "richard";
+		String lastName = "bower";
+		String screenName = "richard bower";
+
+		User user = userFixture.createAnUser(firstName, lastName, screenName);
+
+		userIndexerFixture.searchOnlyOne(firstName);
```
Comment (lmarqs) [inline modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerReindexTest.java]: @luanmaoski, I think we should to delete the documents and check if no one is returned.
Please, check this commit: https://github.com/BryanEngler/liferay-portal/pull/255/commits/b2239c7fe115e0a157cae284664861b4ec187993#diff-86ec1a129f830870ccd023608d0fbf67

File: modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerIndexedFieldsTest.java (hunk 2 above)
Comment (luanmaoski) [reply modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/indexer/test/UserIndexerIndexedFieldsTest.java]: You are right. I will change that. The only field that needs `lowerCase` and isn't sortable is `username`

**Pull request closed without merging**